lib/pgBackRest/Common/Ini.pm

```diff
@@ -42,212 +43,325 @@ use constant INI_KEY_FORMAT                                         => 'backrest
+# headerCheck() - check that version and checksum in header are as expected
+####################################################################################################################################
+sub headerCheck
+{
+    my $self = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $bIgnoreInvalid,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '->headerCheck', \@_,
+            {name => 'bIgnoreInvalid', optional => true, default => false, trace => true},
+        );
+
+    # Eval so exceptions can be ignored on bIgnoreInvalid
+    my $bValid = true;
+
+    eval
+    {
         # Make sure the ini is valid by testing checksum
-        my $strChecksum = $self->get(INI_SECTION_BACKREST, INI_KEY_CHECKSUM);
+        my $strChecksum = $self->get(INI_SECTION_BACKREST, INI_KEY_CHECKSUM, undef, false);
         my $strTestChecksum = $self->hash();
 
-        if ($strChecksum ne $strTestChecksum)
+        if (!defined($strChecksum) || $strChecksum ne $strTestChecksum)
         {
-            confess &log(ERROR, "${strFileName} checksum is invalid, should be ${strTestChecksum} but found ${strChecksum}",
-                         ERROR_CHECKSUM);
+            confess &log(ERROR,
+                "invalid checksum in '$self->{strFileName}', expected '${strTestChecksum}' but found " .
+                    (defined($strChecksum) ? "'${strChecksum}'" : '[undef]'),
+                ERROR_CHECKSUM);
         }
 
         # Make sure that the format is current, otherwise error
         my $iFormat = $self->get(INI_SECTION_BACKREST, INI_KEY_FORMAT, undef, false, 0);
 
-        if ($iFormat != BACKREST_FORMAT)
+        if ($iFormat != $self->{iInitFormat})
         {
-            confess &log(ERROR, "format of ${strFileName} is ${iFormat} but " . BACKREST_FORMAT . ' is required', ERROR_FORMAT);
+            confess &log(ERROR,
+                "invalid format in '$self->{strFileName}', expected $self->{iInitFormat} but found ${iFormat}", ERROR_FORMAT);
         }
 
         # Check if the version has changed
-        if (!$self->test(INI_SECTION_BACKREST, INI_KEY_VERSION, undef, BACKREST_VERSION))
+        if (!$self->test(INI_SECTION_BACKREST, INI_KEY_VERSION, undef, $self->{strInitVersion}))
         {
-            $self->set(INI_SECTION_BACKREST, INI_KEY_VERSION, undef, BACKREST_VERSION);
+            $self->set(INI_SECTION_BACKREST, INI_KEY_VERSION, undef, $self->{strInitVersion});
         }
+
+        return true;
     }
-    else
+    or do
     {
-        $self->numericSet(INI_SECTION_BACKREST, INI_KEY_FORMAT, undef, BACKREST_FORMAT);
-        $self->set(INI_SECTION_BACKREST, INI_KEY_VERSION, undef, BACKREST_VERSION);
-    }
-
-    return $self;
-}
+        # Confess the error if it should not be ignored
+        if (!$bIgnoreInvalid)
```

Comment:
In the next patch, where there are two copies of the ini file, errors must sometimes be ignored while determining which file is correct.